pysubtools/parsers/encodings.py

```diff
@@ -112,16 +112,16 @@ def detect(data, encoding=None, language=None):
     # Reverse order
     encodings.reverse()
     while True:
-        encoding = encodings.pop()
-        if can_decode(data, encoding if not isinstance(encoding, tuple) else
-                      encoding[0]):
+        encoding = encoding_to_use = encodings.pop()
```

Comment:
Naming could be better.

For instance:
```python
encoding_with_confidence = encodings.pop()
if isinstance(encoding_with_confidence, tuple):
  encoding = encoding_with_confidence[0]
else:
  encoding = encoding_with_confidence
  encoding_with_confidence = (encoding, None)
```

Reply:
Just a small FYI (not to fix it, just to avoid code like this in future). I was here reusing `encoding` input parameter. It should be better to be named as `default_encoding`, it is less error prone.